cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/includes/bottom.jsp

```diff
@@ -24,8 +24,8 @@
 </div>  <!-- END #content -->
 
 <c:if test="${empty serviceParam}">
-    <c:set var="serviceParam"
-           value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : ''}"/>
+    <%-- Try to get the service from the login context first. If failed, try to get it from the request parameters. --%>
+    <c:set var="serviceParam" value="&service=${osfLoginContext.isServiceUrl() ? osfLoginContext.getServiceUrl() : param.service ? fn:escapeXml(param.service) : ''}"/>
```

Comment:
When handling certain errors (e.g. rate limiting), the login context may end up with an empty service URL. Alternatively, escape and use the service URL from the request parameter so that bottom links works correctly.